File: hw/src/main/java/com/sumian/sleepdoctor/widget/pay/PayItemGroupView.java
package com.sumian.sleepdoctor.widget.pay;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.View;
import android.widget.LinearLayout;

import com.sumian.sleepdoctor.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by sm
 * on 2018/1/22.
 * desc:
 */

public class PayItemGroupView extends LinearLayout implements PayItemView.OnPayWayCallback {

    @BindView(R.id.wechat_pay_item)
    PayItemView mWechatPayWay;

    @BindView(R.id.alipay_pay_item)
    PayItemView mAlipayWay;

    private OnSelectPayWayListener mOnSelectPayWayListener;

    public PayItemGroupView(Context context) {
        this(context, null);
    }

    public PayItemGroupView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public PayItemGroupView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initView(context);
    }

    private void initView(Context context) {
        ButterKnife.bind(inflate(context, R.layout.lay_pay_item_group, this));
        setGravity(Gravity.CENTER);
        setOrientation(VERTICAL);
        mWechatPayWay.setOnPayWayCallback(this);
        mAlipayWay.setOnPayWayCallback(this);

    }

    public void setOnSelectPayWayListener(OnSelectPayWayListener onSelectPayWayListener) {
        mOnSelectPayWayListener = onSelectPayWayListener;
    }

    @Override
    public void onSelectPayWay(View v) {
        switch (v.getId()) {
            case R.id.wechat_pay_item:
                mAlipayWay.unSelect();
                if (mOnSelectPayWayListener != null) {
                    mOnSelectPayWayListener.onSelectWechatPayWay();
                }
                break;
            case R.id.alipay_pay_item:
                mWechatPayWay.unSelect();
                if (mOnSelectPayWayListener != null) {
                    mOnSelectPayWayListener.onSelectAlipayWay();
                }
                break;
            default:
                break;
        }
    }

    public interface OnSelectPayWayListener {

        void onSelectWechatPayWay();

        void onSelectAlipayWay();
    }
}

File: app/src/main/java/com/sumian/sd/pay/pay/PayItemGroupView.kt
package com.sumian.sd.pay.pay

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.view.View
import android.widget.LinearLayout

import com.sumian.sd.R
import kotlinx.android.synthetic.main.lay_pay_item_group.view.*


/**
 * Created by sm
 * on 2018/1/22.
 * desc:
 */

class PayItemGroupView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : LinearLayout(context, attrs, defStyleAttr), PayItemView.OnPayWayCallback {


    private var mOnSelectPayWayListener: OnSelectPayWayListener? = null

    init {
        initView(context)
    }

    private fun initView(context: Context) {
        View.inflate(context, R.layout.lay_pay_item_group, this)
        gravity = Gravity.CENTER
        orientation = LinearLayout.VERTICAL
        wechat_pay_item.setOnPayWayCallback(this)
        alipay_pay_item.setOnPayWayCallback(this)
    }

    fun setOnSelectPayWayListener(onSelectPayWayListener: OnSelectPayWayListener) {
        mOnSelectPayWayListener = onSelectPayWayListener
    }

    override fun onSelectPayWay(v: View) {
        when (v.id) {
            R.id.wechat_pay_item -> {
                alipay_pay_item.unSelect()
                if (mOnSelectPayWayListener != null) {
                    mOnSelectPayWayListener!!.onSelectWechatPayWay()
                }
            }
            R.id.alipay_pay_item -> {
                wechat_pay_item.unSelect()
                if (mOnSelectPayWayListener != null) {
                    mOnSelectPayWayListener!!.onSelectAlipayWay()
                }
            }
            else -> {
            }
        }
    }

    interface OnSelectPayWayListener {

        fun onSelectWechatPayWay()

        fun onSelectAlipayWay()
    }
}
